asistente_ladm_col/logic/quality_rules/qr_boundary_points_not_covered_by_boundary_nodes.py

```diff
@@ -0,0 +1,221 @@
+            msg = QCoreApplication.translate("QualityRules", "All boundary points are covered by boundary nodes!")
+
+        self.progress_changed.emit(100)
+
+        return QualityRuleExecutionResult(res_type, msg, count)
+
+    @staticmethod
+    def get_boundary_points_not_covered_by_boundary_nodes(db, boundary_point_layer, boundary_layer, point_bfs_layer, id_field):
+        boundary_nodes_layer = GeometryUtils.get_unique_nodes_layer(boundary_layer, id_field)
+
+        # Spatial Join between boundary_points and boundary_nodes
+        spatial_join_layer = processing.run("qgis:joinattributesbylocation",
+                                            {'INPUT': boundary_point_layer,
+                                             'JOIN': boundary_nodes_layer,
+                                             'PREDICATE': [0],  # Intersects
+                                             'JOIN_FIELDS': [db.names.T_ID_F],
+                                             'METHOD': 0,
+                                             'DISCARD_NONMATCHING': False,
+                                             'PREFIX': '',
+                                             'OUTPUT': 'memory:'})['OUTPUT']
+
+        exp_point_bfs = '"{}" is not null and "{}" is not null'.format(db.names.POINT_BFS_T_LC_BOUNDARY_POINT_F,
+                                                                       db.names.POINT_BFS_T_LC_BOUNDARY_F)
+
+        list_point_bfs = [{'boundary_point_id': feature[db.names.POINT_BFS_T_LC_BOUNDARY_POINT_F],
+                           'boundary_id': feature[db.names.POINT_BFS_T_LC_BOUNDARY_F]}
+                          for feature in point_bfs_layer.getFeatures(exp_point_bfs)]
+
+        spatial_join_boundary_point_boundary_node = [{'boundary_point_id': feature[id_field],
+                                                     'boundary_id': feature[id_field + '_2']}
+                                                     for feature in spatial_join_layer.getFeatures()]
+
+        boundary_point_without_boundary_node = list()
+        no_register_point_bfs = list()
+        duplicate_in_point_bfs = list()
+
+        # point_bfs topology check
+        for item_sj in spatial_join_boundary_point_boundary_node:
+            boundary_point_id = item_sj['boundary_point_id']
+            boundary_id = item_sj['boundary_id']
+
+            if boundary_id != NULL:
+                if item_sj not in list_point_bfs:
+                    no_register_point_bfs.append((boundary_point_id, boundary_id))  # no registered in point bfs
+                elif list_point_bfs.count(item_sj) > 1:
+                    duplicate_in_point_bfs.append((boundary_point_id, boundary_id))  # duplicate in point bfs
+            else:
+                boundary_point_without_boundary_node.append(boundary_point_id)  # boundary point without boundary node
+
+        return boundary_point_without_boundary_node, no_register_point_bfs, duplicate_in_point_bfs
```

Comment:
I think we should move this function to GeometryUtils

Reply:
fixed in [1888fd7](https://github.com/SwissTierrasColombia/Asistente-LADM-COL/commit/1888fd77c8479c427aab555a9c7f73bf2027f819)